### PR TITLE
No null extra-deps to be included in stack.yaml files with no extra-deps

### PR DESCRIPTION
**Pull request by Vincibean**
My [previous PR](https://github.com/stevemao/dep-haskell-stack/pull/162) introduced a bug: the parameter `extra-deps` isn't read if the original `stack.yaml` file doesn't contain it, but it is still written, resulting in an "updated" `stack.yaml` file with a `extra-deps: null` line.

This PR aims to fix that by not including an `extra-deps` line if that wasn't part of the `stack.yaml` file in the first place.

Given that my previous PR unfortunately introduced this bug, please don't hesitate to let me know if something in this one looks suspicious.

Here is an example of the old behaviour: https://github.com/Vincibean/Vincibean.github.io/pull/22/files

Here is an example of the new behaviour: https://github.com/Vincibean/test-stack-no-extra-deps/pull/1/files

### CODE REVIEW

File: __tests__/yaml.test.ts [hunk 1]
```diff
@@ -72,6 +73,33 @@ describe('yaml.ts', () => {
...
+    const { doc } = await getStackYaml(
+      `${__dirname}/fixtures/stackNoExtraDeps.yaml`
+    )
+    const extraDeps: Package[] = []
+    const updated = await setExtraDeps(doc, extraDeps)
+    const updatedExtraDeps = await getExtraDeps(updated)
+    expect(updatedExtraDeps).toHaveLength(0)
+  })
+
+  it('should not update any extra dep if the list of updated extra deps is empty', async () => {
```
Comment (stevemao) [inline __tests__/yaml.test.ts:86]: I'm a bit confused with this test. Where is the empty extra deps list?

File: src/yaml.ts [hunk 2]
```diff
@@ -45,13 +45,14 @@ export const setExtraDeps = async (
...
   extraDeps: Package[]
 ): Promise<Document> => {
-  const seq = doc.get('extra-deps') as YAMLSeq
-  // eslint-disable-next-line github/array-foreach
-  extraDeps.forEach((dep, index) => {
-    seq.set(index, `${dep.name}-${dep.version}`)
-  })
-  doc.set('extra-deps', seq)
-
+  if (extraDeps.length && doc.has('extra-deps')) {
```
Comment (stevemao) [inline src/yaml.ts:48]: If I understand you correctly, we only need to check `extraDeps.length`, if true the other condition should be true as well? and vice versa.

**Comment by stevemao** [comment]
Thanks. I left 2 questions.

**Comment by Vincibean** [comment]
Hi @stevemao ! Sorry for the delay. I've addressed your comments; hope it all makes sense.

One thing that may be worth noting: I struggled to find a way to test for the absence of something (in my case: the absence of `extra-deps: null`). In the end I'm not 100% sure that my approach is robust enough, so if you have any suggestion to address this feel free to let me know.

**Review by stevemao — APPROVED** [review]

**Comment by stevemao** [comment]
I think it looks good. Thanks a lot!

**Pull request merged**